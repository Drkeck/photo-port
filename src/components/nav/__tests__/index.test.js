import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Nav from '..';

const catagoties = [
    { name: 'portraits', description: 'Portraits of people in my life' }
]
const mockCurrentCatagory = jest.fn();
const mockSetCurrentCategory = jest.fn();
const mockContactSelected = jest.fn();
const mockSetContactSelected = jest.fn();

afterEach(cleanup)

describe('new nav component', () => {
    it('renders', () => {
        render(<Nav 
        catagoties={catagoties}
        setCurrentCategory={mockSetCurrentCategory}
        currentCatagory={mockCurrentCatagory}
        contactSelected={mockContactSelected}
        setContactSelected={mockSetContactSelected}
        />)
    })

    it('matches Snapshot', () => {
        const {asFragment} = render(<Nav 
            catagoties={catagoties}
            setCurrentCategory={mockSetCurrentCategory}
            currentCatagory={mockCurrentCatagory}
            contactSelected={mockContactSelected}
            setContactSelected={mockSetContactSelected} />)
        expect(asFragment).toMatchSnapshot();
    })
})

describe('the emoji is visable', () => {
    it('inserts the emoji into the h2 element', () => {
        const { getByLabelText } = render(<Nav 
            catagoties={catagoties}
            setCurrentCategory={mockSetCurrentCategory}
            currentCatagory={mockCurrentCatagory}
            contactSelected={mockContactSelected}
            setContactSelected={mockSetContactSelected} />)
        expect(getByLabelText('camera')).toHaveTextContent('ðŸ“¸')
    })
})

describe('links are visible', () => {
    it('inserts text into the links', () => {
      const { getByTestId } = render(<Nav 
        catagoties={catagoties}
        setCurrentCategory={mockSetCurrentCategory}
        currentCatagory={mockCurrentCatagory}
        contactSelected={mockContactSelected}
        setContactSelected={mockSetContactSelected} />);
      expect(getByTestId("link")).toHaveTextContent('Oh Snap!');
      expect(getByTestId("about")).toHaveTextContent('About me');
    });
  })