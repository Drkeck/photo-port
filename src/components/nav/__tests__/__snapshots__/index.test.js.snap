// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new nav component matches Snapshot 1`] = `[Function]`;
